src/main/groovy/com/palantir/gradle/docker/DockerComponent.java

```diff
@@ -13,26 +13,29 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package com.palantir.gradle.docker;
 
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import org.gradle.api.artifacts.DependencyConstraint;
 import org.gradle.api.artifacts.DependencySet;
+import org.gradle.api.artifacts.ExcludeRule;
 import org.gradle.api.artifacts.ModuleDependency;
 import org.gradle.api.artifacts.PublishArtifact;
 import org.gradle.api.attributes.AttributeContainer;
 import org.gradle.api.attributes.Usage;
+import org.gradle.api.capabilities.Capability;
 import org.gradle.api.internal.attributes.ImmutableAttributes;
 import org.gradle.api.internal.attributes.ImmutableAttributesFactory;
 import org.gradle.api.internal.component.SoftwareComponentInternal;
 import org.gradle.api.internal.component.UsageContext;
 import org.gradle.api.model.ObjectFactory;
 
-import java.util.Collections;
-import java.util.LinkedHashSet;
-import java.util.Set;
-
 public class DockerComponent implements SoftwareComponentInternal {
```

Comment:
why do people do this :(

Reply:
Something for later: When implementing interfaces where methods might get added later, I recommend implementing them via a proxy. That way when they add something new, at least you *might* have a chance of not failing (if the new methods are never called).